File: src/main/java/kr/mjc/rachel/spring/di/javaconfig/SonySpeaker.java
package kr.mjc.rachel.spring.di.javaconfig;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class SonySpeaker implements Speaker {
  public SonySpeaker() {
    log.debug("sonySpeaker 인스턴스 생성됨.");
  }

  @Override
  public void volumeUp() {
    log.debug("sonySpeaker volume up.");
  }

  @Override
  public void volumeDown() {
    log.debug("sonySpeaker volume down");
  }
}

File: src/main/kotlin/kr/mjc/jacob/springcore/di/javaconfig/SonySpeaker.kt
package kr.mjc.jacob.springcore.di.javaconfig

import org.slf4j.LoggerFactory

class SonySpeaker : Speaker {

  private val log = LoggerFactory.getLogger(this.javaClass)

  override fun volumeUp() {
    log.debug("sony speaker volume up")
  }

  override fun volumeDown() {
    log.debug("sony speaker volume down")
  }
}